### DISCUSSION TITLE
Make sure only one window opens when you press a key combination, even if you press the keys multiple times.

### DISCUSSION THREAD

**Discussion by mobsenpai** [hyprwm/Hyprland#5627] [Q&A]
For example, consider a keybinding that opens Wofi when I press Super + A. If I accidentally press this key combination multiple times, I end up with multiple Wofi windows that I need to manually close. I've encountered similar issues with other keybindings, such as printscreen, which is set to run grim and slurp. Pressing it twice results in multiple selections appearing on top of each other. I'm looking for a solution that ensures only one window or action is triggered when I press a keybinding, such as focusing the existing Firefox window instead of launching a new one or just launching the application one time if launched already.

**Comment by ordy** [comment]
That would have to be an option then, because you obviously still want the current multiple instance behavior by default.

What most people end up doing is simply prefixing the commands with a `preg -x <app> >/dev/null 2>&1 || <app>` or write a similar wrapper in shell if they need add things like refocus or notifications.

Also some dmenu like tofi even have a config option to prevent that (using a lock file).

**Comment by glxguard** [comment]
Maybe that's not the best variant, but you can just execute `killall wofi || wofi`, so wofi will open and close on one bind
